ios/RCTWebRTC/VideoCaptureController.m

```diff
@@ -71,15 +71,31 @@ -(void)startCapture {
         return;
     }
 
-    [_capturer startCaptureWithDevice:device format:format fps:_fps];
+    // Starting the capture happens on another thread. Wait for it.
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+    [_capturer startCaptureWithDevice:device format:format fps:_fps completionHandler:^(NSError *err) {
+        if (err) {
+            RCTLogError(@"[VideoCaptureController] Error starting capture: %@", err);
+        } else {
+            RCTLog(@"[VideoCaptureController] Capture started");
+        }
+        dispatch_semaphore_signal(semaphore);
+    }];
 
-    RCTLog(@"[VideoCaptureController] Capture started");
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
```

Comment:
Does it run on the WebRTC thread and will block it forever if callback does not happen? I'd say it should give up after some time and log an error.

Reply:
I'm thinking 10 seconds

Reply:
Also: this wait does not happen when permissions are being requested? It's just async stuff?

Reply:
Permissions happen out of here, so no, this won't wait for that.

I don't want to add an arbitrary timeout. The handler will always be called. If it's not, it means we have crashed os the OS is in a bad state, in which case there is nothing we can do to help anyway.

Reply:
ok